File: client/scalaz/src/main/scala/org/openapitools/client/api/SendigDatasetVariable.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SendigDatasetVariable._

case class SendigDatasetVariable (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
description: Option[String],
role: Option[String],
simpleDatatype: Option[String],
core: Option[String],
describedValueDomain: Option[String],
valueList: Option[List[String]],
links: Option[SendigDatasetVariableLinks])

object SendigDatasetVariable {
  import DateTimeCodecs._

  implicit val SendigDatasetVariableCodecJson: CodecJson[SendigDatasetVariable] = CodecJson.derive[SendigDatasetVariable]
  implicit val SendigDatasetVariableDecoder: EntityDecoder[SendigDatasetVariable] = jsonOf[SendigDatasetVariable]
  implicit val SendigDatasetVariableEncoder: EntityEncoder[SendigDatasetVariable] = jsonEncoderOf[SendigDatasetVariable]
}

File: client/clojure/src/cdisc_library_api/specs/sendig_dataset_variable.clj
(ns cdisc-library-api.specs.sendig-dataset-variable
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.sendig-dataset-variable-links :refer :all]
            )
  (:import (java.io File)))


(def sendig-dataset-variable-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :role) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :core) string?
   (ds/opt :describedValueDomain) string?
   (ds/opt :valueList) (s/coll-of string?)
   (ds/opt :_links) sendig-dataset-variable-links-spec
   })

(def sendig-dataset-variable-spec
  (ds/spec
    {:name ::sendig-dataset-variable
     :spec sendig-dataset-variable-data}))
